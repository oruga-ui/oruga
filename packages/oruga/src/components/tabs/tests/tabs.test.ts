import { describe, test, expect, afterEach } from "vitest";
import { enableAutoUnmount, mount } from "@vue/test-utils";

import OTabs from "@/components/tabs/Tabs.vue";

describe("OTabs tests", () => {
    enableAutoUnmount(afterEach);

    test("render correctly", () => {
        const wrapper = mount(OTabs);
        expect(!!wrapper.vm).toBeTruthy();
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.attributes("data-oruga")).toBe("tabs");
        expect(wrapper.html()).toMatchSnapshot();
    });
});
