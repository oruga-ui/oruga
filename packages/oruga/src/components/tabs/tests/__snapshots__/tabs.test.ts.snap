// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OTabs tests > render correctly 1`] = `
"<div class="o-tabs" data-oruga="tabs">
  <nav class="o-tabs__nav o-tabs__nav--default" role="tablist" aria-orientation="horizontal">
    <!--
                @slot Additional slot before tabs
            -->
    <!--
                @slot Additional slot after tabs
            -->
  </nav>
  <section class="o-tabs__content">
    <!--
                @slot Place tab items here
            -->
  </section>
</div>"
`;
